PythonVisualizations/BinaryTreeBase.py

```diff
@@ -0,0 +1,418 @@
+   def createNoneShape(self, node, level, lineCoords, direction, tag):
+      y = lineCoords[3] + (self.CIRCLE_SIZE // 2)
+      xCoord = self.calculateCoordinates(node, level+1, direction)[0]
+
+      shrinkFactor = self.CIRCLE_SIZE // 3
+
+      coords1 = ( xCoord-self.CIRCLE_SIZE, y, 
+                  xCoord+self.CIRCLE_SIZE, y)
+      coords2 = (  shrinkFactor + coords1[0], coords1[1] + shrinkFactor,
+                  -shrinkFactor + coords1[2], coords1[3] + shrinkFactor)
+      coords3 = (  shrinkFactor + coords2[0], coords2[1] + shrinkFactor, 
+                  -shrinkFactor + coords2[2], coords2[3] + shrinkFactor)
+
+      self.canvas.create_line(coords1, tag = tag)
+      self.canvas.create_line(coords2, tag = tag)
+      self.canvas.create_line(coords3, tag = tag)
+
+   # ----------- SETTER METHODS --------------------
+   
+   # create a Node object with key and parent specified
+   # parent should be a Node object
+   def createNode(self, key, parent = None, direction = None):
+      # calculate the level
+      if not parent: level = 0
+      else: level = self.getLevel(parent) + 1                                                
+      
+      # calculate the coords
+      coords = self.calculateCoordinates(parent, level, direction)
+      x,y = coords
+
+      # generate a tag
+      tag = self.generateTag()
+
+      # create the shape and text
+      circle, circle_text = self.createNodeShape(x, y, key, tag)
+      
+      # create the drawable object
+      drawableObj = drawable(key, self.canvas.itemconfigure(circle, 'fill'), *(circle, circle_text))
+      
+      # create the Node object
+      node = Node(drawableObj, coords, tag)
+
+      # increment size
+      self.size += 1
+
+      # add the node object to the internal representation
+      self.setNode(node, direction, parent)
+
+      # draw the lines
+      leftLine, rightLine = self.calculateLineCoordinates(node)
+      self.canvas.create_line(*leftLine, tags = (tag, "left line"))
+      self.canvas.create_line(*rightLine, tags = (tag, "right line"))
+
+      # draw the none symbols
+      self.createNoneShape(node, level, leftLine, Child.LEFT, tag)
+      self.createNoneShape(node, level, rightLine, Child.RIGHT, tag)
+
+      return node 
+
+   # set the node's left child
+   def setLeftChild(self, node, child):
+      index = self.getLeftChildIndex(node)
+      if index != -1:
+         self.nodes[index] = child
+         return index
+      else:
+         return -1
+
+   # set the node's right child
+   def setRightChild(self, node, child):
+      index = self.getRightChildIndex(node)
+      if index != -1:
+         self.nodes[index] = child
+         return index
+      else:
+         return -1
+
+   # set the node's child
+   # returns the index where the child is stored
+   def setNode(self, child, direction = None, parent = None):
+      if not parent:
+         self.nodes[0] = child
+         return 0
```

Comment:
I think this method would be better named as `setChild` or `setChildNode`.

Reply:
Just changed this in 31fac974a96664356e707c6e9d18ae7622f3ed2f